docs/DafnyRef/UserGuide.md

```diff
@@ -1727,7 +1727,31 @@ Detailed description of the `dafny build` and `dafny run` commands and
 the `--input` option (needed when `dafny run` has more than one input file)
 is contained [in the section on command-line structure](#command-line).
 
-### 13.7.2. C\#
+### 13.7.2. Placeholder modules
+To allow defining behavior that depending on which target language Dafny is translated to, requires different Dafny code to be implemented, Dafny has placeholder modules. Here follows an example:
```

Comment:
nit: I'd like to tweak this a little to describe the abstract feature first, then the primary motivation. Something like like "To support customizing behavior differently at compile-time for different contexts, Dafny has placeholder modules... This is particularly useful for defining behavior that depends on which target language Dafny is translated to".

I really like that the feature is generic enough to apply to other use cases so let's not sell it short!

Reply:
In fact I'd promote at least most of this section to the reference manual since it is in fact a language feature, not just a tool feature.

Reply:
> In fact I'd promote at least most of this section to the reference manual since it is in fact a language feature, not just a tool feature.

Not sure where to put it then, there is no "runtime behavior" section, and I think this is a pretty good spot for now.